18-API/src/modules/items/items.controller.ts

```diff
@@ -0,0 +1,112 @@
+import 'reflect-metadata';
+import { NextFunction, Request, Response } from 'express';
+import { injectable, inject } from 'inversify';
+import { BaseController } from '../../common/base.controller';
+import { HTTPError } from '../../errors/http-error.class';
+import { ILogger } from '../../logger/logger.interface';
+import { TYPES } from '../../types';
+import { IFindItemParams, ISortItemParams } from '../../interfaces';
+import { IItemController } from './items.controller.interface';
+import { ValidateMiddleware } from '../../common/validate.middleware';
+import { IConfigService } from '../../config/config.service.interface';
+import { IItemService } from './items.service.interface';
+import { ItemUpdateSatusDto } from './dto/item-update-status.dto';
+import { AuthGuard } from '../../common/auth.guard';
+import { CheckUserRole } from '../../common/checkUserRole.middleware';
+import { EUserRoles } from '../../enum';
+import { ItemCreateDto } from './dto/item-create.dto';
+// middlewares:  [new AuthGuard(), new CheckUserRole([EUserRoles.ADMIN]), new ValidateMiddleware(PromotionUpdateSatusDto)],
+
+@injectable()
+export class ItemController extends BaseController implements IItemController {
+	constructor(
+		@inject(TYPES.ILogger) private loggerService: ILogger,
+		@inject(TYPES.ItemService) private itemService: IItemService,
+		@inject(TYPES.ConfigService) private configService: IConfigService,
+	) {
+		super(loggerService);
+		this.bindRoutes([
+			{
+				path: '/createitem',
```

Comment:
Поправить пути у роутов
```suggestion
				path: '/create-item',
```